server.go

```diff
@@ -496,6 +501,32 @@ func (s *Server) monitorRuntime() {
 	}
 }
 
+// CountOpenFiles on opperating systems that support lsof
+func CountOpenFiles() int {
+	count := 0
+
+	switch runtime.GOOS {
+	case "darwin":
+		fallthrough
+	case "linux":
+		fallthrough
+	case "unix":
+		fallthrough
+	case "freebsd":
+		out, err := exec.Command("/bin/sh", "-c", fmt.Sprintf("lsof -p %v", os.Getpid())).Output()
```

Comment:
you might explore using the -b option off of lsof.  It prevents the call from blocking.  It will change the format  of the Output a tad but would protect against a hung process